### PR TITLE
Retry tls and key decryptions in systemd units

### PR DESCRIPTION
**Pull request by MarcelMue**
This is a first step towards improvements for E2E, retries here prevent the unit to block the whole machine if the first request to quay fails.

### CODE REVIEW

**Comment by xh3b4sd** [comment]
Do you want to pair here tomorrow a little bit to get the e2e tests more stable?

**Comment by MarcelMue** [comment]
I will take care of cluster api and post some logs to confirm it works.

I simply retried builds before until the pulling failed once and then looked at the units logs to see the retry message.

File: service/controller/legacy/v26/templates/cloudconfig/decrypt_keys_assets_script.go [hunk 1]
```diff
@@ -1,6 +1,25 @@
...
+    local max_retries=10
+    local delay=15
+    while true; do
+        "$@" && break || {
+        if [[ $n -lt $max_retries ]]; then
+            ((n++))
+            echo "Command failed. Attempt $n/$max_retries:"
+            sleep $delay;
+        else
+            fail "The command has failed after $n attempts."
```
Comment (kopiczko) [inline service/controller/legacy/v26/templates/cloudconfig/decrypt_keys_assets_script.go]: `fail`?

File: service/controller/legacy/v26/templates/cloudconfig/decrypt_keys_assets_script.go [hunk 2]
```diff
@@ -1,6 +1,25 @@
...
+        fi
+        }
+    done
+}
```
Comment (kopiczko) [inline service/controller/legacy/v26/templates/cloudconfig/decrypt_keys_assets_script.go]: This is higher level shit.

**Comment by corest** [comment]
I don't think this will work. The current issue can't be solved with retries

**Comment by xh3b4sd** [comment]
Please elaborate on this. Marcel saw that it solves the problem apparently.

**Comment by MarcelMue** [comment]
I think @corest is thinking about china - this here fixes a flap in e2e.

2 Problems, same piece of code.

File: service/controller/legacy/v26/templates/cloudconfig/decrypt_keys_assets_script.go (hunk 1 above)
Comment (MarcelMue) [reply service/controller/legacy/v26/templates/cloudconfig/decrypt_keys_assets_script.go]: Good catch, lost something while refactoring, I'll provide a fix.

**Comment by MarcelMue** [comment]
I will provide a new PR after testing again because these units changed when we fixed china.

**Pull request closed without merging**